var/spack/repos/builtin/packages/magma/package.py

```diff
@@ -186,21 +186,40 @@ def post_install(self):
     def cache_test_sources(self):
         """Copy the example source files after the package is installed to an
         install test subdirectory for use during `spack test run`."""
-        self.cache_extra_test_sources([self.test_src_dir])
+        cache_extra_test_sources(self, [self.test_src_dir])
 
-    def test(self):
+    def test_all_c(self):
+        """Run benchmark tests"""
         test_dir = join_path(self.test_suite.current_test_cache_dir, self.test_src_dir)
         with working_dir(test_dir, create=False):
-            pkg_config_path = "{0}/lib/pkgconfig".format(self.prefix)
+            pkg_config_path = self.prefix.lib.pkgconfig
             with spack.util.environment.set_env(PKG_CONFIG_PATH=pkg_config_path):
+
                 make("c")
-                self.run_test("./example_sparse", purpose="MAGMA smoke test - sparse solver")
-                self.run_test(
-                    "./example_sparse_operator", purpose="MAGMA smoke test - sparse operator"
-                )
-                self.run_test("./example_v1", purpose="MAGMA smoke test - legacy v1 interface")
-                self.run_test("./example_v2", purpose="MAGMA smoke test - v2 interface")
-                if "+fortran" in self.spec:
-                    make("fortran")
-                    self.run_test("./example_f", purpose="MAGMA smoke test - Fortran interface")
+                tests = [
+                    ("example_sparse", "sparse solver"),
+                    ("example_sparse_operator", "sparse operator"),
+                    ("example_v1", "legacy v1 interface"),
+                    ("example_v2", "v2 interface"),
+                ]
+
+                for test in tests:
+                    with test_part(
+                        self, f"test_all_c_{test[0]}", purpose=f"MAGMA smoke test - {test[1]}"
+                    ):
+                        exe = which("./" + test[0])
+                        exe()
+
                 make("clean")
+
+    def test_fortran(self):
+        """MAGMA smoke test - Fortran interface"""
+        if "+fortran" not in self.spec:
+            raise SkipTest("Package must be installed with +fortran")
+
+        test_dir = join_path(self.test_suite.current_test_cache_dir, self.test_src_dir)
```

Comment:
This should also be within the environment context (see current lines 195-196) as it was in the original.

Current outputs:
```
==> [2024-08-09-18:03:42.077236] test: test_fortran: MAGMA smoke test - Fortran interface
==> [2024-08-09-18:03:42.078408] 'make' '-j16' 'fortran'
Package magma was not found in the pkg-config search path.
Perhaps you should add the directory containing `magma.pc'
to the PKG_CONFIG_PATH environment variable
Package 'magma', required by 'virtual:world', not found
Package magma was not found in the pkg-config search path.
Perhaps you should add the directory containing `magma.pc'
to the PKG_CONFIG_PATH environment variable
Package 'magma', required by 'virtual:world', not found
Package magma was not found in the pkg-config search path.
Perhaps you should add the directory containing `magma.pc'
to the PKG_CONFIG_PATH environment variable
Package 'magma', required by 'virtual:world', not found
```